assets/js/state/sagas/softwareUpdates.js

```diff
@@ -0,0 +1,28 @@
+import { get } from 'lodash';
+import { put, call, takeEvery } from 'redux-saga/effects';
+import { getSoftwareUpdates } from '@lib/api/softwareUpdates';
+
+import {
+  FETCH_SOFTWARE_UPDATES,
+  startLoadingSoftwareUpdates,
+  setSoftwareUpdates,
+  setEmptySoftwareUpdates,
+  setSoftwareUpdatesErrors,
+} from '@state/softwareUpdates';
+
+export function* fetchSoftwareUpdates({ payload: { hostId } }) {
+  yield put(startLoadingSoftwareUpdates());
+
+  try {
+    const response = yield call(getSoftwareUpdates, hostId);
+    yield put(setSoftwareUpdates({ hostId, softwareUpdates: response.data }));
+  } catch (error) {
+    yield put(setEmptySoftwareUpdates());
```

Comment:
I guess we should continue keeping track of the host id also on failure, so empty only for the specific host. Otherwise we end up in emptying the whole state, for all the hosts.